DOC: update pandas cheat sheet with a third page (fixes #40680)

Shoestring42 opened the pull request:
- [x] closes #40680
- [x] added 3rd page to the cheat sheet including more details on plotting, frequently used options, input/output formats and more.
- [x] added `.info()`, `.memory_usage()`, and `.dtypes()` to 'Summarize Data' on page 2 and rearranged the page to fill the gap left by the old plotting section.

Shoestring42 committed:
DOC: update pandas cheat sheet with a third page (fixes #50680)

Dr-Irv reviewed (changes requested):
Thanks for this contribution.  Here are some things to address, all on page 3:

- In the examples, when you have single quotes, make sure they are using `'` rather than the curly left and curly right quote that Powerpoint sticks in there.  See the plot examples where that happens.
- Be consistent in coloring.  pandas methods should be in blue, arguments in black.  For the various charts, the colors are backwards.
- It's worth mentioning in the plot section that you can combine arguments (e.g. `subplots`, `title`).  But be careful with some of those arguments, as I think `stacked` may only apply to bar charts.  
- You have the same description for `df.plot(kind='hist')` and `df.plot(kind='area')` which doesn't seem right
- In the "Changing Type" section, you have a header "More on datetime" that is very small.  I think it should be it's own section, i.e., larger and describe that if you have date-time oriented data, then on a series `s`, you can use the `dt` accessor to get various parts of the date and time.  Doesn't seem to fit under the "Changing Type category"
- Under mapping, I would make the examples and description about using `map()` and `apply()` on `Series`, not `DataFrame`
- The description of a pivot table isn't correct.  The key difference with `pivot_table()` as compared to `pivot()` is that you can supply a function that aggregates the values.  So I'd remove the picture (not clear how you indicate the agg func pictorially), but broaden the description.  Also, the method is `df.pivot_table()` not `pd.pivot_table()`.
- In the input/output section, I think it is important to say that you can write CSV, and Excel files.  Also, the `to_xxx` functions are wrong - they are of the form `df.to_xxx(filepath)`  And the `read` functions should have a `pd.` in front of them
- In the options section, put the functions in blue
- Put the clipboard stuff in the Input/Output section.  Also, it is `pd.read_clipboard()` not `df.read_clipboard()`
-

Shoestring42 committed:
DOC: update cheat sheet based on review feedback (40680)

Shoestring42 commented:
Thanks for the feedback, this is my first time contributing to anything so thanks for being so constructive. I think I've addressed all the issues raised. Changes have been pushed so let me know if anything else needs updating!

Dr-Irv reviewed (changes requested):
Thanks for the update.  Here are some additional things I found:

- On page 3, in the description following each item, capitalize the first letter of the first word.  E.g., under "Series String Operations", it is correct, but under "Changing type", it is not.
- Under `df.plot.area()`, change wording to be "Plot an area graph..."  (change "a" to "and")
- Under "Datetime", reword paragraph:  "With a Series containing data of type datetime, the dt accessor is used to get various components of the datetime values:"
- Under "Mapping", in the description, change "DataFrame" to "DataFrame or Series"
- Under "Mapping", change the second example to be `df.apply(lambda s: s.max() - s.min(), axis=1)` with a description of "Returns a Series with the difference of the maximum and minimum values of each row of the DataFrame"
- The `to_clipboard()` example should be `df.to_clipboard()`
- Under the Options category, change lead sentence to say "control how pandas behaves" rather than "how DataFrames behave"
- Under the Options category, it's worth mentioning (probably at the top) that options can be queried and set via `pd.options._option_name_ where _option_name_ is the name of an option, and then illustrate via an example, e.g., `pd.options.display.max_rows = 20`

That's what I see right now.  This is a great contribution, and it's getting close.

You'll also have to add a PDF version to the PR.

Also, can you update https://github.com/pandas-dev/pandas/tree/main/doc/cheatsheet/README.md below the first table with the sentence:  "The English version has additional material that is not in the versions in other languages."

Shoestring42 committed:
DOC: 3rd draft: update cheat sheet (40680)

Shoestring42 commented:
Thanks again, I've updated the powerpoint according to the feedback, made the pdf version, and editted the README.md file as well.

Dr-Irv reviewed (changes requested):
One last thing and then we should be good to go.

In the "Frequently Used Options" section, instead of `_option_name_`, use `option_name`, but put that in italics.  I can't show how that would look here, but the font should be the same, but just italicized.   It should look like this:
![image](https://github.com/user-attachments/assets/6c37693c-715b-4a1a-bfd3-ba194608461a)

Then you change it in the next sentence as well.  

Also, the example with `max_rows`, should not have `Pd` capitalized.

And the description of that should say "Set the `display.max_rows` option to 20", where `display.max_rows` is in the monospaced font.

Dr-Irv commented:
@Shoestring42 looking forward to having you address my last comment here:  https://github.com/pandas-dev/pandas/pull/61378#pullrequestreview-2809938830

Shoestring42 committed:
Italicised 'option_name' and fixed the max_rows example

Shoestring42 commented:
changes have been pushed. Sorry it took so long, exam period has just started at uni so i've been busy revising. Thanks for your patience.

Dr-Irv reviewed (approved):
Thanks @Shoestring42 .  Great contribution!
